Go/syntax_test_go.go

```diff
@@ -66,84 +66,87 @@ You may have to disable Go-specific linters when working on this file.
     //go:generate one two three
 // ^ -comment -meta -punctuation
 //  ^^ punctuation.definition.comment.go
-//  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.annotation.go comment.line.go
+//  ^^^^^^^^^^^^^^ comment.line.go meta.annotation.go
```

Comment:
@deathaxe I'd like to understand why `meta.annotation.go` should not reach to the end of the line and if it doesn't why it does reach into the space that separates it from the `meta.annotation.parameters.go`.

Reply:
It doesn't need to not do so, but this was the state of the art before my commit. So I just maintained it. Same applies to the space in fron to  `meta.annotation.parameters`.

I'd be very fine with both being changed.

One aspect to considder though is annotations being highlighted using `background` by some color schemes.

![grafik](https://user-images.githubusercontent.com/16542113/156783067-71f2b407-58de-4fe9-9b68-757aad31f77b.png)

By scoping `\n` `meta.annotation` those background colors would span up to the very right edge of a view. Is it what we want?

Reply:
Sorry, but the image doesn't help me understand that answer.

I was asking two things.

1. Why not make to the end of the text like so:
    ```
        //lint:ignore U1000 Reason.
    //  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.go meta.annotation.go
    ```
    This comes down to what the semantics of a `meta.annotation.go` are and something that I may not be understanding. But to my mind, the entire directive comment is a meta annotation.

2. If they do not make it to the end of the line as above, why cover the space after the directive name like so
    ```
        //lint:ignore U1000 Reason.
    //  ^^^^^^^^^^^^^^ comment.line.go meta.annotation.go
    ```
    rather than like so
    ```
        //lint:ignore U1000 Reason.
    //  ^^^^^^^^^^^^^ comment.line.go meta.annotation.go
    ```
   The former having ugly background highlighting as described above.

Reply:
> 1. Why not make to the end of the text like so:

Because we want a single `meta.annotation` on stack at any point, only. Each character is scoped `meta.annotation`. `U1000 Reason.` are scoped `meta.annotation as well, but have assigned another "subscope" `parameters`. The API function `view.match_selector(pt, `meta.annotation`) would return `True` for each character.

```
    //lint:ignore U1000 Reason.
//  ^^^^^^^^^^^^^^ meta.annotation
//                ^^^^^^^^^^^^^ meta.annotation.parameters
```

Previously it was:

```
    //lint:ignore U1000 Reason.
//  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.annotation
//                ^^^^^^^^^^^^^ meta.annotation.parameters
```

So `U1000 Reason.` were scoped `meta.annotation meta.annotation.parameters`. This is the kind of scope duplication or "stacking of scopes", we want to avoid here.

> 2. If they do not make it to the end of the line as above, why cover the space after the directive name ...

The `[ ]?` at the end of https://github.com/sublimehq/Packages/blob/91daefae708d63281c43a65e14867286ebb9238e/Go/Go.sublime-syntax#L135 causes `meta.annotation.parameters` which is matched by https://github.com/sublimehq/Packages/blob/91daefae708d63281c43a65e14867286ebb9238e/Go/Go.sublime-syntax#L143-L144 to start at `U1000` not covering the space in front of it.

I just truncated the `meta.annotation.go` without touching boundaries of `meta.annotation.parameters`. And I applied this scheme to all the other patterns for consistency reasons. The space should (must) be in- or excluded the same way in all patterns.

_It would be absolutly correct to change that behavior, if we do so for all patterns in the same way._


> The former having ugly background highlighting as described above.

I can't really follow, because a color scheme should address `meta.annotation` rather than `meta.annotation.go` to ensure to correctly apply to all syntaxes. If it does so, an assigned background would span the whole line (as my screenshot above illustrates). As `meta.variable` is assigned to `ignore` only, any color applied to that scope would not apply to the space after `ignore`.


```
    //lint:ignore U1000 Reason.
//  ^^^^^^^^^^^^^^ meta.annotation
//         ^^^^^^ meta.variable.function
//               ^ - meta.variable.function
```

Reply:
Thanks. I'll think about this. Part of the issue wit the image is that the contrast is just not visible; I can't tell what it is saying at all.

Reply:
![grafik](https://user-images.githubusercontent.com/16542113/156877412-77e430d0-76ea-47de-b3b0-044c8b79480b.png)

Reply:
The test marks are fine, just the contrast. This is better though, thanks.

I think that given the definition of the derectives includes a punctuation space between the directive and the parameters that space+ should probably be its own capture like the colon is. I'll send a PR tomorrow with my tweaks.

Reply:
Had some time to think more clearly about this and I agree, this is good.